lib/chef/provider/aws_rds_subnet_group.rb

```diff
@@ -0,0 +1,84 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsRdsSubnetGroup < Chef::Provisioning::AWSDriver::AWSProvider
+
+  def create_aws_object
+    converge_by "create new RDS Subnet Group #{new_resource.name} in #{region}"do
+      driver.create_db_subnet_group(desired_options)
+    end
+  end
+
+  def destroy_aws_object(object)
+    converge_by "delete RDS Subnet Group #{new_resource.name} in #{region}" do
+      driver.delete_db_subnet_group(db_subnet_group_name: new_resource.db_subnet_group_name)
+    end
+  end
+
+  def update_aws_object(object)
+    updates = required_updates(object)
+    if ! updates.empty?
+      converge_by updates do
+        driver.modify_db_subnet_group(desired_options)
+      end
+    end
+  end
+
+  def desired_options
+    opts = {}
+    opts[:db_subnet_group_name] = new_resource.db_subnet_group_name
+    opts[:db_subnet_group_description] = new_resource.db_subnet_group_description
+    opts[:subnet_ids] = new_resource.subnet_ids
+    opts[:tags] = tag_hash_to_array(new_resource.aws_tags) if new_resource.aws_tags
+    opts
+  end
+
+  # Given an existing object, return an array of update descriptions
+  # representing the updates that need to be made.
+  #
+  # If no updates are needed, an empty array is returned.
+  #
+  def required_updates(object)
+    ret = []
+    if desired_options[:db_subnet_group_description] != object[:db_subnet_group_description]
+      ret << "  set group description to #{desired_options[:db_subnet_group_description]}"
+    end
+
+    if ! xor_array(desired_options[:subnet_ids], subnet_ids(object[:subnets])).empty?
+      ret << "  set subnets to #{desired_options[:subnet_ids]}"
+    end
+
+    if ! (desired_options[:aws_tags].nil? || desired_options[:aws_tags].empty?)
```

Comment:
I don't want to block this PR on being able to update tags.  I think we could merge it without update support now and file an issue to improve it later.

But for that to work we'll have to make sure `aws_tags` gets unset on an update, otherwise https://github.com/chef/chef-provisioning-aws/blob/master/lib/chef/provisioning/aws_driver/aws_provider.rb#L245 will raise an exception